aspnetcore/fundamentals/minimal-apis/openapi.md

```diff
@@ -77,52 +77,143 @@ ASP.NET Core provides the [`Microsoft.AspNetCore.OpenApi`](https://www.nuget.org
+
+:::code language="csharp" source="endpoint-description-summary/7.0-samples/OpenAPISummary/Program.cs" id="summary_and_description2" highlight="2-3":::
+
+Alternatively, the summaries can be set directly on the OpenAPI annotation.
+
+```csharp
+app.MapGet("/todoitems2", async (TodoDb db) => await db.Todos.ToListAsync())
+    .WithOpenApi(operation => new(operation)
+    {
+        Summary = "This is a summary",
+        Description = "This is a description"
+    });
+```
+
 ### Exclude Open API description
 
 In the following sample, the `/skipme` endpoint is excluded from generating an OpenAPI description:
 
 [!code-csharp[](7.0-samples/WebMinAPIs/Program.cs?name=snippet_swag2&highlight=20-21)]
 
+### Mark an API as obsolete
+
+To mark an endpoint as obsolete, set the `Depercated` property on the OpenAPI annotation.
+
+```csharp
+app.MapGet("/todos", async (TodoDb db) => await db.Todos.ToListAsync())
+    .WithOpenApi(operation => new(operation)
+    {
+        Deprecated = true
+    });
+```
+
 ### Describe response types
 
-The following example uses the built-in result types to customize the response:
+OpenAPI supports providing an description of the responses returned from an API. Minimal APIs supports three strategies for setting the response type of an endpoint:
 
-[!code-csharp[](samples/todo/Program.cs?name=snippet_getCustom)]
+* Via the `Produces` extension method
+* Via the `ProducesResponseType` attribute
+* By returning `TypedResults` from your application
 
-### Add operation IDs to Open API
+The `Produces` extension method can be used to add `Produces` metadata to an endpoint. When no parameters are provided, the extension method will populate metadata for the targeted type under a `200` status code and an `application/json` content type.
 
-[!code-csharp[](samples/todo/Program.cs?name=snippet_name)]
+```csharp
+app
+    .MapGet("/todos", async (TodoDb db) => await db.Todos.ToListAsync())
+    .Produces<IList<Todo>>();
+```
 
-### Add tags to the Open API description
+Leveraging `TypedResults` in the implementation of an endpoint's route handler will automatically include the response type metadata for the endpoint. For example, the code below will automatically annotate the endpoint with a response under the `200` status code with an `application/json` content type.
 
-The following code uses an [OpenAPI grouping tag](https://swagger.io/docs/specification/grouping-operations-with-tags/):
+```csharp
+app.MapGet("/todos", async (TodoDb db) =>
+{
+    var todos = await db.Todos.ToListAsync());
+    return TypedResults.Ok(todos);
+});
+```
 
-[!code-csharp[](samples/todo/Program.cs?name=snippet_grp)]
+#### Set responses for ProblemDetails
 
-### Add endpoint summary or description
+When setting the response type for endpoints that may return a ProblemDetails response, the `ProducesProblem` extension method can be used to provide the annotation or the `TypedResults.Problem` return type.
```

Comment:
suggestion:
"the `ProducesProblem` extension method or the `TypedResults.Problem` return type can be used to provide the annotation."